src/main/java/org/mule/extension/ftp/internal/connection/FtpFileSystem.java

```diff
@@ -101,9 +101,9 @@ private static String resolveBasePath(String basePath, FTPClient client) {
     copyCommand = new FtpCopyCommand(this, client);
     createDirectoryCommand = new FtpCreateDirectoryCommand(this, client);
     deleteCommand = new FtpDeleteCommand(this, client);
-    listCommand = new FtpListCommand(this, client);
     moveCommand = new FtpMoveCommand(this, client);
     readCommand = new FtpReadCommand(this, client);
+    listCommand = new FtpListCommand(this, client, (FtpReadCommand) readCommand);
```

Comment:
why this movement?

Reply:
I need the readCommand to be instanciated before creating the listCommand. Another way would be to make the getReadCommand() method public and the list command would use it instead of receiving it in the constructor.

Reply:
ohh sorry, I didn't saw that. my bad